index.js

```diff
@@ -476,39 +472,88 @@ io.on('connection', (socket) => {
 }
 
-async function reminderMessage(location) {
-    console.log("Reminder message being sent");
-    let session = await db.getMostRecentSession(location); // Gets the updated state for the chatbot
+async function reminderMessage(sessionid) {
+  console.log("Reminder message being sent");
+  let session = await db.getSessionWithSessionId(sessionid); // Gets the updated state for the chatbot
+  if(session.chatbot_state == STATE.STARTED) {
+    //Get location data
+    var location = session.locationid;
     locationData = await db.getLocationData(location)
-    if(session.chatbot_state == 'Started') {
-        //send the message
-        await sendTwilioMessage(locationData.twilio_number, session.phonenumber, `This is a reminder to check on the bathroom`)
-        session.chatbot_state = 'Waiting for Response';
-        await db.saveChatbotSession(session);
-    }
-    //else do nothing
+    //send the message
+    await sendTwilioMessage(locationData.twilio_number, session.phonenumber, `This is a reminder to check on the bathroom`)
+    session.chatbot_state = STATE.WAITING_FOR_RESPONSE;
+    let chatbot = new Chatbot(session.sessionid, session.locationid, session.chatbot_state, session.phonenumber, session.incidenttype, session.notes);
+    await db.saveChatbotSession(chatbot);
+  }
+  //else do nothing
 }
 
+async function fallbackMessage(sessionid) {
+  console.log("Fallback message being sent");
+  let session = await db.getSessionWithSessionId(sessionid); // Gets the updated state for the chatbot
+  if(session.chatbot_state == STATE.WAITING_FOR_RESPONSE) {
+    console.log("Fallback if block");
+    let locationData = await db.getLocationData(session.locationid)
+    console.log(`fallback number is:  ${locationData.fallback_phonenumber}`)
+    console.log(`twilio number is:  ${locationData.twilio}`)
+    await sendTwilioMessage(locationData.twilio_number, locationData.fallback_number,`An alert to check on the washroom at ${locationData.location_human} was not responded to. Please check on it`)
+  }
+//else do nothing
+} 
+
+
 //Heartbeat Helper Functions
+
+async function checkHeartbeat() {
+
+ for(let i = 0; i < locations.length; i++){
+  let location = await db.getLocationData(locations[i]);
+  // Query raw sensor data to transmit to the FrontEnd
+  let XeThruData = await redis.getLatestXeThruSensorData(location.locationid);
+
+  // Check the XeThru Heartbeat
+  let currentTime = moment();
+  let latestXethru = moment(XeThruData.timestamp, "x");
+  let XeThruDelayMillis = currentTime.diff(latestXethru);
+
+  if(XeThruDelayMillis > XETHRU_THRESHOLD_MILLIS && !location.xethru_sent_alerts) {
+    console.log(`XeThru Heartbeat threshold exceeded; sending alerts for ${location.locationid}`)
+    await db.updateSentAlerts(location.locationid, true)
+    sendAlerts(location.locationid)
+  }
+  else if((XeThruDelayMillis < XETHRU_THRESHOLD_MILLIS) && location.xethru_sent_alerts) {
```

Comment:
This looks like it's a copy-paste of `index.js` lines 339-342. If it needs to be in both places, could it be put into a function that is called by both places? That will help if something needs to change in the logic then it will only need to be changed in one place.